### DISCUSSION TITLE
Unexpected problems with @types/jest

### DISCUSSION THREAD

**Discussion by tim-rohrer** [coder/code-server#4246] [Q&A] [answered]
I’m asking this here because I’m using the same code repo on my laptop that I am in the VPS `workspace` for my `code-server` instance, and I am not having this problem in VS Code on the laptop. I admit it could be something I’ve forgotten in VS Code settings, but I’ve not yet found it.

I’m getting this error with jest spec files:

```
describe("Unit test", () => {
  it("Does something", () => {
    expect(1).toBe(true);
  });
});
```

```
Cannot find name 'describe'. Do you need to install type definitions for a test runner? Try `npm i --save-dev @types/jest` or `npm i --save-dev @types/mocha` and then add 'jest' or 'mocha' to the types field in your tsconfig.
```

To be clear, the type definitions are installed.

Is there something unique required for `code-server` to handle these types? I’m scrubbing docs I can find on the Internet, but as I said, I don’t have this problem in VSCode so I’m confused.

Edits======

Here is my eslint config:

```{
    "env": {
        "browser": false,
        "es2020": true,
        "jest": true
    },
    "settings": {
        "import/resolver": {
          "node": {
            "paths": ["src"],
            "extensions": [".js", ".ts"]
          }
        }
    },
    "extends": [
        "airbnb-base"
    ],
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 11,
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
        "import/extensions":0,
        "no-unused-vars": ["error", { "argsIgnorePattern": "next" }],
        "lines-between-class-members": ["error", "always", { "exceptAfterSingleLine": true }],
        "quotes": ["error", "double"]
    }
}
```

**Comment by tim-rohrer** [comment, marked as answer]
This can be disregarded. I tried to delete the question, but don’t see a way to do that.

Later in the weekend I ended up being able to reproduce the issues with VSCode, although I’m not sure how. I do seem to be having issues with ESLint and Typescript, but these is probably better suited for another forum. Although I’m not real experienced, this isn’t my first project, and I’m a bit baffled by what is causing me to have trouble now.

Either way, it doesn’t appear to be related to `code-server`.

**Reply by jsjoeio** [reply]
Sorry to hear you're having trouble! It's always a struggle getting all these tools to play together nicely. I usually follow one of [these guides](https://dev.to/sam_piggott/setting-up-typescript-with-eslint-prettier-for-visual-studio-code-1e3h) when setting all this up.

If you're still having trouble after that, I would try disabling all VS Code extensions and deleting your `settings.json` to see if that helps you isolate the problem. Good luck! We're around if you need more help.
